### PR TITLE
feat: create employee request for nhs

### PR DESCRIPTION
**Pull request by svetavedmed**
edenlabllc/ehealth.api#4480

### CODE REVIEW

File: __schema__/employeeRequests.graphql [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  "Day diapason when employeeRequest was created."
+  insertedAt: Date
+  "EmployeeRequest status. The value should be present in the `EMPLOYEE_REQUEST_STATUS` dictionary."
+  status: String
+  "Legal entity in which employee works."
+  legalEntityId: UUID
+}
+
+"""
+Methods to use when ordering `Employee`.
```
Comment (progsmile) [inline __schema__/employeeRequests.graphql]:
```suggestion
Methods to use when ordering `EmployeeRequest`.
```

File: __schema__/employeeRequests.graphql [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+"""
+enum EmployeeRequestOrderBy {
+  "Sort by full name in ascending order."
+  FULL_NAME_ASC
```
Comment (progsmile) [inline __schema__/employeeRequests.graphql]: Suppose it would be hard to sort entities by nested json fields, for such case index is must. @kiurchv what do you think?

File: __schema__/employeeRequests.graphql [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  "Birth date of person"
+  birthDate: Date!
+  "Party's email and login for a user"
+  email: String!
+  "Type of employee, is connected with user'd role. The value should be present in the `EMPLOYEE_TYPE` dictionary."
+  employeeType: String!
+  "First name of a person"
+  firstName: String!
+  "Day diapason when employeeRequest was created."
+  insertedAt: Date
```
Comment (progsmile) [inline __schema__/employeeRequests.graphql]: insertedAt: `DateTime`
```suggestion
  insertedAt: DateTime
```

Comment (progsmile) [reply __schema__/employeeRequests.graphql]: Let's also add `updatedAt: DateTime` and group them in bottom

File: __schema__/employeeRequests.graphql [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  "National person identifier."
+  taxId: String!
+}
+
+"""
+Inputs that should be send in order to create employee request. User must have a scope **employee_request:create**
+"""
+input CreateEmployeeRequestInput {
+  "Employee working address"
+  division: ID
```
Comment (progsmile) [inline __schema__/employeeRequests.graphql]:
```suggestion
  divisionId: ID
```

File: __schema__/parties.graphql [hunk 5]
```diff
@@ -44,12 +45,54 @@ type Party implements Node {
...
+  "the date when document was issued"
+  issuedAt: Date
+  "document expiration date"
+  expirationDate: Date
+}
+
+"""
+Inputs that should be send in order to create party.
+"""
+input CreatePartytInput {
```
Comment (PavelVesnin) [inline __schema__/parties.graphql]:
```suggestion
input CreatePartyInput {
```

File: __schema__/parties.graphql [hunk 6]
```diff
@@ -44,12 +45,54 @@ type Party implements Node {
...
+  "Party birth date."
+  birthDate: Date!
+  "Party identifier documents"
+  documents: [CreatePartyDocumentInput]!
+  "Party email, user login"
+  email: CreateUserInput!
```
Comment (PavelVesnin) [inline __schema__/parties.graphql]: email it's an object with field email. What for?
`email: {email: string}`

File: __schema__/employeeRequests.graphql [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+  "Type of employee, is connected with user'd role. The value should be present in the `EMPLOYEE_TYPE` dictionary."
+  employeeType: String!
+  "First name of a person"
+  firstName: String!
+  "Second name of a person"
+  secondName: String
+  "Surname of a person"
+  lastName: String!
+  "Legal entity in which employee will work."
+  legalEntityId: LegalEntity!
```
Comment (PavelVesnin) [inline __schema__/employeeRequests.graphql]: I believe it should be:
```suggestion
  legalEntityId: ID!
```

File: __schema__/employeeRequests.graphql [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+  "Flag to show whether person rejected to have taxId"
+  noTaxId: Boolean
+  "Status of employee request. The value should be present in the `EMPLOYEE_REQUEST_STATUS` dictionary."
+  status: String!
+  "National person identifier."
+  taxId: String!
+  "Date and time when record was created."
+  insertedAt: DateTime
```
Comment (PavelVesnin) [inline __schema__/employeeRequests.graphql]:
```suggestion
  insertedAt: DateTime!
```

File: __schema__/employeeRequests.graphql [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+"""
+input EmployeeRequestFilter {
+  "Future user's login, email."
+  email: String
+  "Day diapason when employeeRequest was created."
+  insertedAt: Date
+  "EmployeeRequest status. The value should be present in the `EMPLOYEE_REQUEST_STATUS` dictionary."
+  status: String
+  "Legal entity in which employee works."
+  legalEntityId: UUID
```
Comment (PavelVesnin) [inline __schema__/employeeRequests.graphql]:
```suggestion
  legalEntityId: ID
```

File: __schema__/employeeRequests.graphql [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+"""
+Inputs that should be send in order to create employee request. User must have a scope **employee_request:create**
+"""
+input CreateEmployeeRequestInput {
+  "Employee working address"
+  divisionId: ID
+  "Type of employee, is connected with user'd role. The value should be present in the `EMPLOYEE_TYPE` dictionary."
+  employeeType: String!
+  "Personal information of employee"
+  party: CreatePartytInput!
```
Comment (PavelVesnin) [inline __schema__/employeeRequests.graphql]:
```suggestion
  party: CreatePartyInput!
```

File: __schema__/employeeRequests.graphql [hunk 11]
```diff
@@ -72,14 +72,14 @@ type EmployeeRequest implements Node {
...
-  "Legal entity in which employee will work."
-  legalEntityId: LegalEntity!
   "Flag to show whether person rejected to have taxId"
   noTaxId: Boolean
   "Status of employee request. The value should be present in the `EMPLOYEE_REQUEST_STATUS` dictionary."
   status: String!
   "National person identifier."
   taxId: String!
+  "Legal entity in which employee will work."
+  legalEntity: LegalEntity!
```
Comment (PavelVesnin) [inline __schema__/employeeRequests.graphql]: employee request has `legal_entity_id`, not legal_entity map

File: __schema__/employeeRequests.graphql [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+  "Date and time when record was created."
+  insertedAt: DateTime
+  "Date and time when record was updated"
+  updatedAt: DateTime!
+}
+
+"""
+Inputs that should be send in order to create employee request. User must have a scope **employee_request:create**
+"""
+input CreateEmployeeRequestInput {
```
Comment (progsmile) [inline __schema__/employeeRequests.graphql]: The request data is signed, isn't it?
```
input CreateEmployeeRequestInput {
   signedContent: SignedContent!
}
```

Comment (progsmile) [reply __schema__/employeeRequests.graphql]: @svetavedmed

File: __schema__/employeeRequests.graphql (hunk 11 above)
Comment (progsmile) [reply __schema__/employeeRequests.graphql]: It refers to entity, so should be named as `legalEntity`, not `legalEntityId`

File: __schema__/employeeRequests.graphql (hunk 9 above)
Comment (progsmile) [reply __schema__/employeeRequests.graphql]: That should be uuid, because it will be under signed content, and would be inconvenient to decode

File: __schema__/index.graphql [hunk 13]
```diff
@@ -569,6 +569,13 @@ type Mutation {
   createEmployeeRequest(
     input: CreateEmployeeRequestInput!
```
Comment (progsmile) [inline __schema__/index.graphql]: Suppose we should replace this input with `CreateEmployeeRequestSignedInput`
and remove `createEmployeeRequestSigned` mutation

File: commitlint.config.js [hunk 14]
```diff
@@ -0,0 +1 @@
+module.exports = { extends: ["@commitlint/config-conventional"] };
```
Comment (kiurchv) [inline commitlint.config.js]: Is this file really needed?

File: __schema__/employeeRequests.graphql [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+  "Date and time when record was created."
+  insertedAt: DateTime!
+  "Date and time when record was updated"
+  updatedAt: DateTime!
+}
+
+"""
+Inputs that should be send in order to create employee request via signed data.
+"""
+input CreateEmployeeRequestInput {
```
Comment (kiurchv) [inline __schema__/employeeRequests.graphql]: Input types should go before node types

File: commitlint.config.js (hunk 14 above)
Comment (progsmile) [reply commitlint.config.js]: Yes
